mcs/class/Microsoft.Build.Utilities/Microsoft.Build.Utilities/ProcessWrapper.cs

```diff
@@ -74,6 +74,7 @@ public new void Start ()
 		public void WaitForOutput (int milliseconds)
 		{
 			CheckDisposed ();
+			WaitForExit (milliseconds);
 			WaitHandle.WaitAll (new WaitHandle[] { endEventOut, endEventErr, endEventExit }, milliseconds);
```

Comment:
@ludovic-henry those are two different things we wait for. I think @radical's code is correct. If it waits up to 200ms (for instance) for the exit, then so be it - it is still a tiny amount of time compared to Process startup/teardown times, and the time required to process the output. Out of the two above, too, waiting for exit is more crucial since it is more commonplace to check for process.ExitCode which, should the process is not done, will throw as we can see in the OP. This code is/should be geared more for correctness than speed.